### ISSUE TITLE
Expose `RequestOptions` for HLC methods causing noisy logs/tracing

### ISSUE THREAD

**Issue by tg-msft** [Azure/azure-sdk-for-net#25626]
The upcoming `Azure.RequestOptions` gives us the chance to expose [`ErrorOptions`](https://docs.microsoft.com/en-us/dotnet/api/azure.erroroptions?view=azure-dotnet) from our convenience methods.  We're not planning to do this everywhere, but there have been a handful of cases where the noise of an expected exception degraded the observability/diagnosability experience for customers.  This was discussed in detail in the ArchBoard meetings at https://github.com/Azure/azure-sdk/issues/3391.  This work item tracks enabling the first set of experiences for HLC libraries.

#### Foundational
- [x] Add `RequestOptions` - https://github.com/Azure/autorest.csharp/issues/1666
- [x] Support `RequestOptions` via the AutoRest's HLC generator - https://github.com/Azure/autorest.csharp/issues/1734

#### Use internally for both sync/async overloads:
- [ ] (Azure.Storage.Blobs) Blob Download - https://github.com/Azure/azure-sdk-for-net/issues/9934
- [x] (Azure.Storage.Blobs) Blob Exists - https://github.com/Azure/azure-sdk-for-net/issues/18592

#### Expose publicly both sync/async overloads:
- [ ] (Microsoft.Azure.WebJobs.Extensions.Storage) Receipt Manager - https://github.com/Azure/azure-sdk-for-net/issues/21511

#### Could either expose publicly or add an `If(Not)Exists` method to use internally for both sync/async overloads:
- [ ] (Azure.Search.Documents) Get Document - https://github.com/Azure/azure-sdk-for-net/issues/17445
- [ ] (Azure.Security.KeyVault.*) Delete - https://github.com/Azure/azure-sdk-for-net/issues/25125
- [x] (Azure.Data.Tables) CreateIfNotExists, CreateTableIfNotExists, GetEntitiy - https://github.com/Azure/azure-sdk-for-net/issues/16251
- [ ] (Azure.Data.AppConfiguration) Create - https://github.com/Azure/azure-sdk-for-net/issues/14802#issuecomment-996976716

**Comment by longtimedeveloper** [comment]
@tg-msft can you please add both 'Azure.Data.Tables.GetEntity' and 'Azure.Data.Tables.GetEntityAsync'? 

Thank you!!

**Comment by tg-msft** [comment]
All of our sync/async APIs are required to have parity as validated by [our analyzers](https://github.com/Azure/azure-sdk-tools/blob/main/src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/Descriptors.cs#L33), but I added a comment about it applying to both sync/async methods of all the affected APIs for good measure.

**Comment by pakrym** [comment]
I think some of these don't need a RequestOptions overload.

For example, in https://github.com/Azure/azure-sdk-for-net/issues/18592 the fix is fully internal. The method needs to tell the pipeline that 404 is not an error in all cases.

**Comment by tg-msft** [comment]
Yes @pakrym, this was meant to track everything blocked on `RequestOptions`, but I just split it into three lists to make it clearer which would expose `RequestOptions`, which would merely make use of it, and which we could do either way.

**Comment by annelo-msft** [comment]
@pakrym, we are exploring how to address this without exposing `RequestOptions` in HLC public APIs.

**Comment by annelo-msft** [comment]
The Azure.Core and Generator work for this have been completed.

**Comment by heaths** [comment]
@annelo-msft I talked with @ShivangiReja about generating internal DPG protocol methods to add to our existing KV HLC, but it doesn't sound like quite what I want (we don't generate the core libs currently). Can you tell me which of the "new DPG" SDKs is about to ship with this I could use as an example of the expected signature to add manually?

We want to add a couple methods to each of the core KV clients that allow customers to, say, fetch a secret without throwing an exception on a 404 without adding an explicit method to do *just* that. IIRC, it was @KrzysztofCwalina or @tg-msft that recommended the `RequestContext` (or `RequestOptions`) over that.

**Comment by annelo-msft** [comment]
@heaths could you share links to the REST endpoints of the methods you'd like to add?  It is a fairly straightforward conversion from REST endpoint to protocol method, happy to share info on that.

**Comment by heaths** [comment]
Since we have a thread going on internally that seems to have progressed a bit further, let's continue this conversation there. 🙂

**Comment by Archomeda** [comment]
Is there any target date for this? I keep hitting a lot of warning logs for various methods like CreateIfNotExists, DeleteIfExists, UpsertEntity, etc. For now I'm resorting to just up the Azure.Core logging to error, but I would prefer that these warnings would just not show up at all for expected results.

**Comment by annelo-msft** [comment]
@Archomeda, the foundational pieces have shipped.  Which library are you interested in addressing warning logs in?

**Comment by Archomeda** [comment]
@annelo-msft This is regarding the Azure.Data.Tables and Azure.Storage.Blobs libraries.

**Comment by diligent176** [comment]
OMG, will someone please fix this **409 error** in Table Storage...   **10 years**!!

#109

**Comment by schafersm-fsa** [comment]
I have been following this issue for a few months now as I am having the same problem that is outlined in the following issue: [#28084](https://github.com/Azure/azure-sdk-for-net/issues/28084)

With the release of Azure.Data.Tables 12.7.1 I was expecting the failed dependency errors in app insights when calling TableClient.CreateIfNotExistsAsync to no longer show but this is not the case.

Am I missing something here?

**Comment by christothes** [comment]
> CreateIfNotExistsAsync

Could you clarify which dependency call is failing and with which status code?

**Comment by schafersm-fsa** [comment]
@christothes - It is Table.Create with a status code of 409.

**Comment by heaths** [comment]
Unchecking https://github.com/Azure/azure-sdk-for-net/issues/17445 because it wasn't actually completed. The bot closed it with no other action taken.

**Comment by heaths** [comment]
@schaabs @christothes @AlexanderSher should Core also do something to this effect for the shared challenge-based auth such that the initial 401s aren't adversely tracing? Or is that warranted (to trace)?
